### ISSUE TITLE
work together?

### ISSUE THREAD

**Issue by segler-alex** [y20k/transistor#94]
hi,

i really like the style you choose for your app!
i am working for quite some time on the following projects:
http://www.radio-browser.info
https://github.com/segler-alex/RadioDroid
are you interested in work together, to combine our efforts?
i thought i will come only that far alone, so i am searching for people/programmers who also want a free radio app to happen.
it would be nice to hear from you

greetings from vienna
alex

**Comment by y20k** [comment]
Hi there,

Thank you for contacting me. With regards to Transistor I immediately can think of two ways how to integrate your radio directory.

1. I could add an option for users to submit a station (or all stations) to your directory 
2. I could add a station browser to help users with station discovery

There are some issues to resolve, though.

- **Licenses a:** Your code of radio-browser.info is licensed under GPL. That is great because in case your site goes offline - I could in theory set up my own instance. But: Under which license is the content? Can I copy it too? Creative Commons would be helpful here, I guess.
- **Licenses b**: RadioDroid published is under GPL too. I do not know legal stuff. But can I use the code you put on GitHub in my app. Transistor is licensed under MIT.

> i will come only that far alone

In the past **I did reject suggestions to implement third party** radio directories (see #51). The basic rationale behind it is still the same. I do not want to be tied to any third party stuff. One of the reasons for that is, that I am doing this part time and I do hesitate to take the risks involved with relying on third party stuff. That is my take on "i will come only that far alone"  / on knowing my limits as a one person shop.

> i am searching for people/programmers who also want a free radio app to happen.

I am not sure that you wanted to suggest that, but there is definitely the possibility to put our expertise together and build a new radio app from scratch. I am a passionate radio listener and I would love to radio to stay relevant in the times of Spotify and others. I also am a free software advocate.

There is a problem though(, which is basically an aspect of my personality). I think I am not the best "player" when it comes to team work. I have strong views on what is right what is and wrong, when it comes to user experience questions. The big plus for me in being the only maintainer of Transistor is that I can control everything.

Anyway. When you were suggesting a collaboration, what did you have in mind?

Best regards
Christian

PS. If you ever come to Stuttgart lets have a beer.

**Comment by spaetz** [comment]
Hi there, sorry, I normaly don't do that, but let me my opinion:
I do love transistor, and feel that it does many UI choices right. I am using it  lot. On the other hand I like the radiobrowser directory which is how radioditectories should be (I hate eg. Vtuner.com both in terms of UI and proprietarinrss), but I am not so fond of the radiodroid client UI.

Can I suggest a way to collaborate? Transistor could include a link to the radiobrowser directory on the web (transistor only makes sense when you are online anyway). When you then click on a radio stream in radiobrowser, the resulting .mp3,.ogg, or .p3u file could be offered to be opened/added by Transistor.

This only requires loose collaboration (for now) and whenever one of you quits, it will be easy to change things. Transistor could perhaps have a "rate/submit this stream to radiobrowser" context menu entry, which calls a radiobrowser url that offers to add the radio station if it does not exist yet or to rate/tag an existing station.

P.S
 I'd like to buy both of you a beer for your work and whenever you come to Hamburg, I'll happily drink it with you. :-)

**Comment by y20k** [comment]
Hi there, 

your idea sounds compelling, because adding a link is VERY easy. But I think there must be a more elegant way to integrate the radiobrowser directory. And I think that placing a weblink within the UI is very uncommon and would be seen as bad app design.

I do not want to dismiss the idea too fast, but I need some time to think about it.

Best regards
Christian

PS. I actually am from time to time in Hamburg :-) So I might come back to your beer offer.

**Comment by segler-alex** [comment]
i don't think you got me email, so i am writing here again :)
i made the data completely free and extractable. i make a backup every day automatically from the mariadb. now everybody can download it and play arround with it on a private server.
every kind of use of the database is cool. i am still hoping to find someone to continue radiodroid, so that i can start implementing new uses if the api in other programs, i am still hoping to get into clementine media player or vlc some day.
so if you want to continue radiodroid, its yours :) i want to do the server stuff anyway.
i just did it to create an android app, that makes use of the whole api. but if you think radiodroid is beyound repair :) a fresh start is also a good idea.
the thing i imagine is to get as big as wikipedia, but for online radios. and be as open as wikipedia or even more :)
the bottom line is, anything you feel you want to contribute to a better radio app

**Comment by y20k** [comment]
> i don't think you got my email

That is true. I was about to close this issue yesterday, because I thought you did not answer :-)

> hoping to find someone to continue radiodroid

Transistor and Radiodroid are conceptually different. Radiodroid focus seems to be on browsing and discovering radio stations. Transistor does not have any features that help with discovery. It's focus is on simplicity and having a small fixed set of stations. Both approaches are valid, both approaches cater to different audiences. It is not impossible to create an app that would combine both approaches, but at the moment it seems to be a task that is too big for me. 

> if you think radiodroid is beyond repair :)

Radiodroid and Transistor are great little apps - far from being beyond repair! But to combine both approaches only bits and pieces of both apps could be re-used. A new radio-app needs to be built from the ground up. To do that I lack time and motivation. I have ideas for new apps that I would honestly try to realize before doing a restart on doing a radio app.

I like Radiodroid. It would be sad, if you have to stop development. It seems like you have accomplished so much already. 

Anyway. I will keep an eye on radio-browser.info when considering new features. I like your approach and I am glad that you contacted me and did draw my attention to your project. But I cannot fill the role as a new / or additional maintainer of Radiodroid.

PS. IMHO: Feature-wise a Mini-Player like in [android-UniversalMusicPlayer](https://github.com/googlesamples/android-UniversalMusicPlayer) could increase the usability of Radiodroid.

**Comment by y20k** [comment]
I am closing this one for now. As soon as there is a more concrete plan to integrate radio-browser.info I will open a new ticket here. Thanks again for contacting me.

**Comment by y20k** [comment]
@segler-alex FYI the current version of Transistor ([v4](https://github.com/y20k/transistor/releases/tag/v4.0.0)) has simple a radio-browser.info integration. Just a search box, but it fits nicely with the app.
